### ISSUE TITLE
Indexer crashes

### ISSUE THREAD

**Issue by gerroon** [go-gitea/gitea#5324]
Indexer crashes with the lines below

I am using Gitea Version: 34695f4 on Debian Testing

2018/11/12 11:37:06 [.../runtime/asm_amd64.s:1333 goexit()] [E] updateRepoIndexer: exit status 128 - fatal: git cat-file: could not get object info

2018/11/12 11:37:06 [I] Done populating the repo indexer with existing repositories
2018/11/12 11:37:07 [.../runtime/asm_amd64.s:1333 goexit()] [E] updateRepoIndexer: exit status 128 - fatal: Not a valid object name 33af0e1f6c425959dd41b96752e61f50949e0118
57c601d489a0901efaa5ded23b724bd5fb7f5af2

2018/11/12 11:37:22 [.../runtime/asm_amd64.s:1333 goexit()] [E] updateRepoIndexer: exit status 128 - fatal: Not a valid object name d23a53c74af7601d45f90b182bcde035ef0d691c
6ddd2549edfd65ec21fe54efae21e963b83fba1e

**Comment by techknowlogick** [comment]
Which version of Git are you using?

**Comment by gerroon** [comment]
git --version
git version 2.19.1 

Debian Buster X64

**Comment by tYYGH** [comment]
I have a similar problem: same error message, same status 128. **And Gitea is then taking up 44% of my whole RAM!**
This is after changing from:
* git 2.18.0(-1)
* gitea 1.5.1(-1)

to:
* git 2.19.1(-1)
* gitea 1.5.3(-3)

on Archlinux, with [my Ansible configuration](https://yalis.fr/git/yves/home-server/src/branch/master/roles/ihmgit_back/tasks/main.yml#L51) applied to [`app.ini.sample` from upstream Gitea](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/gitea#n64).

Please ask for details if you need more.

**Comment by xoxys** [comment]
Same here on RHEL:

`updateRepoIndexer: exit status 128 - error: unable to find XXX
fatal: git cat-file XXX: bad file`

- git version 1.8.3.1
- gitea 1.5.3

**Comment by markusamshove** [comment]
Same after updating from `1.5.1` to `1.6.0`:

* SLES 12
* git version 2.12.3
* gitea 1.6.0

**Comment by laoshaw** [comment]
Same, using 1.6.0, debian 9 64bit, git 2.17.1

2018/11/30 22:56:10 [.../runtime/asm_amd64.s:1333 goexit()] [E] updateRepoIndexer: exit status 128 - fatal: git cat-file 9d7337a3b028095abbb0ee3dec460544d019d1f6: bad file

**Comment by ll911** [comment]
same, using 1.6.1, rhel 7.5 X86_64bit, git 2.14.1
```
2018/12/17 15:06:09 [...routers/repo/http.go:391 serviceRPC()] [E] fail to serve RPC(upload-pack): exit status 128 - {[102 97 116 97 108 58 32 84 104 101 32 114 101 109 111 116 101 32 101 110 100 32 104 117 110 103 32 117 112 32 117 110 101 120 112 101 99 116 101 100 108 121 10] 0 [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0] 0}
2018/12/17 17:29:40 [.../runtime/asm_amd64.s:1333 goexit()] [E] updateRepoIndexer: exit status 128 - fatal: git cat-file: could not get object info
```

**Comment by gerroon** [comment]
The indexer seems to be a bit useless with big repos :(

**Comment by tYYGH** [comment]
As things stand, it is _also_ useless with little repos, because you have to disable it to keep it from eating all your RAM :-/

**Comment by laoshaw** [comment]
indexing and searching are essential to code related work, any way to improve this in gitea, as far as I can tell, this is the only obvious weak area at the moment.

**Comment by lunny** [comment]
@laoshaw Yes, that needs more work.

**Comment by tYYGH** [comment]
Does someone know the current status of this issue?
Also, what are the consequences of switching the indexer off? I wish to understand if this issue is minor or major, as far as I am concerned at least…

**Comment by apricote** [comment]
@tYYGH If you switch the Repo Indexer off, you will not be able to search for code in Gitea anymore.

**Comment by gerroon** [comment]
I am wondering about the indexer situation as well. Has there been any improvements?

**Comment by gerroon** [comment]
@stale you suck

**Comment by TBK** [comment]
Here is a full Drone CI Cloud buildlog running `timeout -s ABRT 20m make test-sqlite` documenting the issue - https://cloud.drone.io/alpinelinux/aports/7389/3/1

**Comment by 6543** [comment]
is this fixed now?

**Comment by sapk** [comment]
I will have a look but I remember a PR for a buffered `git cat-file` implementation.

**Comment by guillep2k** [comment]
@sapk @6543 It's probably fixed in #7711 and others.

**Comment by sapk** [comment]
@guillep2k Yes, you greatly improve indexer code and others have greatly contribute to git related performances.

I will close this issue then. 
Please feel free to re-open an issue (and link this one) if the problem is still existing.
